includes/lib/class-faster-image-b52f1a8-faster-image.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+require_once( AMP__DIR__ . '/includes/lib/class-faster-image-b52f1a8-invalid-image-exception.php' );
+require_once( AMP__DIR__ . '/includes/lib/class-faster-image-b52f1a8-image-parser.php' );
+require_once( AMP__DIR__ . '/includes/lib/class-stream-17b32f3-stream.php' );
+require_once( AMP__DIR__ . '/includes/lib/class-stream-17b32f3-stream-buffer-too-small-exception.php' );
+
+/**
+ * FasterImage - Because sometimes you just want the size, and you want them in
+ * parallel!
+ *
+ * Based on the PHP stream implementation by Tom Moor (http://tommoor.com)
+ * which was based on the original Ruby Implementation by Steven Sykes
+ * (https://github.com/sdsykes/fastimage)
+ *
+ * MIT Licensed
+ *
+ * @version 0.01
+ */
+class Faster_Image_B52f1a8_Faster_Image
+{
+    /**
+     * The default timeout
+     *
+     * @var int
+     */
+    protected $timeout = 4;
+
+	/**
+	 * Backup user agent
+	 *
+	 * @var string
+	 */
+	protected $user_agent = 'wp-amp, v' . AMP__VERSION;
```

Comment:
I actually thought about doing this when I initially made the change, but I considered the way most other user agents work. Web browsers don't include anything in their user agents that indicate who the particular user is or where the requested resources are going. I was concerned that by doing so for WP-AMP, we could be putting WordPress publishers who hot link to images they don't host (of which I imagine there are many) at a disadvantage, e.g. by facilitating the blocking of their site from accessing the images via their user agent. Still, I suppose that is better than the possibility of the WP-AMP user agent being blocked altogether. I've added the site url and made it filterable.